tests/integration/azure_blob/test_azure_blob.py

```diff
@@ -74,34 +75,60 @@
     reason="AZURE_CONTAINER is not set",
 )
 class TestAzureBlob:
+    assets_path = os.path.join(os.path.dirname(os.path.dirname(__file__)), "assets")
+    profile_name = "azure_blob_ds_profile"
+    test_dir = "test_mlrun_azure_blob"
+    run_dir = f"{test_dir}/run_{uuid.uuid4()}"
+    bucket_name = config["env"].get("AZURE_CONTAINER", None)
+    test_file = os.path.join(assets_path, "test.txt")
+
     @classmethod
     def setup_class(cls):
-        cls.profile_name = "azure_blob_ds_profile"
-        cls.test_dir = "test_mlrun_azure_blob"
-        cls._bucket_name = config["env"].get("AZURE_CONTAINER", None)
-        cls.test_file = here / "test.txt"
         with open(cls.test_file) as f:
             cls.test_string = f.read()
+        cls._azure_fs = None
 
-    def teardown_method(self):
-        test_dir = f"{self._bucket_name}/{self.test_dir}"
-        if not self._azure_fs:
+    @classmethod
+    def teardown_class(cls):
+        test_dir = f"{cls.bucket_name}/{cls.test_dir}"
+        if not cls._azure_fs:
             return
-        if self._azure_fs.exists(test_dir):
-            self._azure_fs.delete(test_dir, recursive=True)
-            logger.debug("test directory has been cleaned.")
+        if cls._azure_fs.exists(test_dir):
+            cls._azure_fs.delete(test_dir, recursive=True)
+            logger.debug("test directory has been deleted.")
+
+    def teardown_method(self, method):
+        for auth, auth_list in AUTH_METHODS_AND_REQUIRED_PARAMS.items():
+            if auth.startswith("env"):
+                for env_parameter in auth_list:
+                    if config["env"].get(env_parameter, None):
+                        os.environ[env_parameter] = config["env"].get(env_parameter)
+
+    @classmethod
+    def create_fs(cls, storage_options):
+        #  Create filesystem object only once - for teardown_class method.
```

Comment:
```suggestion
        # Create filesystem object only once
```
Pretty sure we create it for the tests, not the teardown 😛